deepchem/data/datasets.py

```diff
@@ -458,8 +464,17 @@ def create_dataset(shard_generator, data_dir=None, tasks=[], verbose=True):
     for shard_num, (X, y, w, ids) in enumerate(shard_generator):
       basename = "shard-%d" % shard_num
       metadata_rows.append(
-          DiskDataset.write_data_to_disk(data_dir, basename, tasks, X, y, w,
-                                         ids))
+          DiskDataset.write_data_to_disk(
+            data_dir,
+            basename,
+            tasks,
+            X,
+            y,
+            w,
+            ids,
+            X_is_sparse))
+      if shard_num % 10 == 0:
+        print("Shard #"+str(shard_num)+" | # generated per minute: ", shard_num * 60 / (time.time()-time1))
```

Comment:
not sure if this is desired, but it sure as hell is useful for featurizing large datasets.

Reply:
We need to re-investigate logging and use a logging library in deepchem.  I would leave it in for now though.